client/state/domains/management/actions.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE,
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST,
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST_FAILURE,
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST_SUCCESS,
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_UPDATE,
+	DOMAIN_MANAGEMENT_WHOIS_RECEIVE,
+	DOMAIN_MANAGEMENT_WHOIS_REQUEST,
+	DOMAIN_MANAGEMENT_WHOIS_REQUEST_FAILURE,
+	DOMAIN_MANAGEMENT_WHOIS_REQUEST_SUCCESS,
+	DOMAIN_MANAGEMENT_WHOIS_SAVE,
+	DOMAIN_MANAGEMENT_WHOIS_SAVE_FAILURE,
+	DOMAIN_MANAGEMENT_WHOIS_SAVE_SUCCESS,
+	DOMAIN_MANAGEMENT_WHOIS_UPDATE,
+} from 'state/action-types';
+
+/**
+ * Returns an action object to be used in signalling that a cached domains
+ * contact details object has been received.
+ *
+ * @param   {Object}   cacheData	cached contact details object
+ * @returns {Object}   Action object
+ */
+export function receiveContactDetailsCache( cacheData ) {
+	return {
+		type: DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE,
+		cacheData
```

Comment:
Actually, worth noting that in Calypso Redux it's more common to inline the fields in the top-level object:

```js
{ type: ROUTE_SET, path: '/' }
```

or, for certain action types and payloads, to fold data under `data`:

```js
{ type: 'FOO', data }
```

the latter would suit this instance, as the nature of the data (a cache) is indicated by the action type.